neuraxle/base.py

```diff
@@ -838,12 +828,13 @@ def handle_fit_transform(self, data_container: DataContainer, context: Execution
         :param context: execution context
         :return: tuple(fitted pipeline, data_container)
         """
-        self.is_invalidated = True
+        data_container, context = self._will_process(data_container, context)
+        data_container, context = self._will_fit_transform(data_container, context)
 
-        new_self, out = self.fit_transform(data_container.data_inputs, data_container.expected_outputs)
-        data_container.set_data_inputs(out)
+        new_self, data_container = self._fit_transform_data_container(data_container, context)
```

Comment:
Note that this forces to always return a new_self which **must** always be overwrited. This would be worth documenting at many different places in many docstrings.

Reply:
there is always a documented return that says fitted self